libsolidity/ast/AST.cpp

```diff
@@ -374,6 +374,29 @@ std::vector<std::pair<ASTPointer<IdentifierPath>, std::optional<Token>>> UsingFo
 	return ranges::zip_view(m_functionsOrLibrary, m_operators) | ranges::to<vector>;
 }
 
+std::string StructDefinition::encodeType() const
+{
+	std::string str = name() + "(";
+	std::set<std::string> nested; // std::set enables duplicates elimination and ordered enumeration
+	for (size_t i = 0; i < m_members.size(); i++)
+	{
+		str += i == 0 ? "" : ",";
+		str += m_members[i]->type()->canonicalName() + " " + m_members[i]->name();
+		if (m_members[i]->type()->category() == Type::Category::Struct) {
+			Declaration const* declaration = m_members[i]->type()->typeDefinition();
+			if (StructDefinition const* structDef = dynamic_cast<StructDefinition const*>(declaration)) {
```

Comment:
Probably we could rely on this check only, avoiding checking category, but not sure relying logic on a `dynamic_cast` is a good practice.